airbyte-integrations/connectors/source-monday/unit_tests/test_graphql_request_options_provider.py

```diff
@@ -0,0 +1,29 @@
+from source_monday import GraphQLRequestOptionsProvider
+
+
+def test_get_request_params(mocker):
+    schema = {
+        "key_a": {
+            "type": ["null", "string"],
+            "properties": {
+                "key_a_1": {
+                    "type": ["null", "string"],
+                    "properties": {
+                        "key_a_2": {"type": ["null", "string"]}
+                    }
+                }
+            }
+        },
+        "key_b": {"type": ["null", "string"]},
+        "key_c": {"type": ["null", "string"]},
+    }
+    mocker.patch.object(GraphQLRequestOptionsProvider, "get_schema_root_properties", return_value=schema)
+    provider = GraphQLRequestOptionsProvider(
+        limit="{{ options['items_per_page'] }}",
+        options={"name": "items", "items_per_page": 100}
+    )
+    assert provider.get_request_params(
+        stream_state=mocker.MagicMock(),
```

Comment:
could you replace these mocks with empty lists or dicts for instance? mocks are very tolerant to types/signature errors

Reply:
There is no any reason to discuss about `stream_state` and `stream_slice`:
- I do not see how it can affect the test
- fields are not a part of the test
- `get_request_params` does not use it at all

Reply:
After some thinking I decided that maybe MagicMock is too much for the args and we don't need most of functionality it proposes. So we can switch to Mock or just basic types as you suggested by the same reasons from comment above